pkg/spec/spec.go

```diff
@@ -30,12 +30,13 @@ type Function struct {
 
 // FunctionSpec contains func specification
 type FunctionSpec struct {
-	Handler  string `json:"handler"`
-	Function string `json:"function"`
-	Runtime  string `json:"runtime"`
-	Type     string `json:"type"`
-	Topic    string `json:"topic"`
-	Deps     string `json:"deps"`
+	Handler  string   `json:"handler"`
+	Function string   `json:"function"`
+	Runtime  string   `json:"runtime"`
+	Type     string   `json:"type"`
+	Topic    string   `json:"topic"`
+	Deps     string   `json:"deps"`
+	Env      []string `json:"env"`
```

Comment:
Don't extend this with an ever-growing list of explicit attributes - use an `api.PodTemplateSpec`, that gets copied into the generated `Deployment` and gives a clear path for supporting all the other pod/container attributes.  In particular this would effectively replace the Env type with the usual `[]api.EnvVar` which also allows for environment variables from ConfigMap/Secret/downward-api/etc.
You'll need to document how any overlapping generated portions (image, etc) are merged with the PodTemplateSpec (just override settings on the first container, and leave any additional sidecars as-is).